compiler+runtime/src/cpp/jank/codegen/llvm_processor.cpp

```diff
@@ -1083,6 +1112,64 @@ namespace jank::codegen
     return ctx->builder->CreateLoad(ctx->builder->getPtrTy(), global);
   }
 
+  llvm::Value *
+  llvm_processor::gen_var_root(obj::symbol_ref qualified_name, var_root_kind kind) const
+  {
+    auto it(ctx->var_root_globals.find(qualified_name));
+    if(it != ctx->var_root_globals.end())
+    {
+      /* If the var-root already exists, and is part of a normal var-deref, create a load in the IR
+       * Else if part of the special cases, just return the value of the var-root directly
+       */
+      if(kind == var_root_kind::global_init)
+      {
+        return ctx->builder->CreateLoad(ctx->builder->getPtrTy(), it->second);
+      }
+      return it->second;
+    }
+    auto const name(util::format("var_root_{}", munge(qualified_name->to_string())));
+    auto const var_root(create_global_var(name));
+    ctx->module->insertGlobalVariable(var_root);
+    auto &global(ctx->var_root_globals[qualified_name]);
+    global = var_root;
+
+    /* When generating a var-root in a context where we are already calling var_bind_root, we can just return the global var-root */
+    if(kind == var_root_kind::binded_def)
+    {
+      return global;
+    }
+    /* Directly derefing the var in the jank_load function */
```

Comment:
This is not a complete sentence. It lacks a subject.

But, more importantly, it's just not easy to read and understand.

Compare it to all of my comments, either in this file, or on your PR.